src/log.c

```diff
@@ -119,23 +119,36 @@ ly_errapptag(const struct ly_ctx *ctx)
 }
 
 API struct ly_err_item *
-ly_err_new(LY_LOG_LEVEL level, LY_ERR no, LY_VECODE vecode, char *msg, char *path, char *apptag)
+ly_err_new(LY_ERR *ecode, LY_VECODE vecode, char *path, char *apptag, const char *err_msg, ...)
```

Comment:
Interesting solution for the memory allocation problem but is it really worth it since it will, well, never happen? Meaning remove this parameters and if it does happen, the return `LY_ERR` value will not match the error in this structure?

Reply:
It was actually just taken from ly_err_msg_create(), but yeah, your suggestion makes sense and improves/simplifies the usage.